### PR TITLE
Feat(dashboard): Display latest ingresses with statusses

### LINKED ISSUE #26: Update frontend to display 20 latest Archived Ingress Messages
Show in table

### LINKED ISSUE #27: Ability to filter on multiple: Device IDs, Pipeline IDs, StepLog Tracing IDs
Update Filter struct in HTTP Transport layer and update PSQL store to filter on new filter properties

### LINKED ISSUE #28: Update frontend ingress messages to display steps and pipeline information
For every tracing ID fetch the relevant information from pipeline service and device service and tracing service in one go by utilizing array id filters

example:
/api/pipelines?id=...&id=...&id=...&id=...

### PR DESCRIPTION
**Pull request by TimVosch**
Closes #28 
Closes #27 
Closes #26

### CODE REVIEW

File: services/tracing/migrations/20230821113939_initial_tracing.up.sql [hunk 1]
```diff
@@ -15,7 +15,7 @@ CREATE TABLE steps (
     tracing_id VARCHAR(255) NOT NULL,
     step_index BIGINT NOT NULL,
     steps_remaining BIGINT NOT NULL,
-    start_time timestamptz NOT NULL,
+    start_time timestamptz(3) NOT NULL,
```
Comment (JEFFTheDev) [inline services/tracing/migrations/20230821113939_initial_tracing.up.sql:18]: what does the 3 do?

File: services/tracing/tracing/infra/store_psql.go [hunk 2]
```diff
@@ -103,13 +104,13 @@ func (s *stepStore) QueryTraces(filter tracing.Filter, r pagination.Request) (*p
 func (s *stepStore) GetStepsByTracingIDs(tracingIds []string) ([]tracing.EnrichedStep, error) {
 	q := sq.Select(
 		"tracing_id",
-		"device_id",
+		"COALESCE(device_id, 0)",
```
Comment (JEFFTheDev) [inline services/tracing/tracing/infra/store_psql.go:107]: what does coalesce here do?

**Review by JEFFTheDev — CHANGES_REQUESTED** [review]

File: services/dashboard/infra/api_stores.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+		devicesEndpoint:   devicesEndpoint,
+	}
+}
+
+func (s *SensorBucketAPI) ListIngresses() ([]ingressarchiver.ArchivedIngressDTO, error) {
+	res, err := http.Get(s.ingressEndpoint)
+	if err != nil {
+		return nil, err
+	}
+	if res.StatusCode != 200 {
```
[the remark this answers is not in the document]
Comment (TimVosch) [reply services/dashboard/infra/api_stores.go:50]: Fixed

File: services/dashboard/infra/api_stores.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+		fmt.Printf("could not decode ingresses response: %v", err)
+		return nil, err
+	}
+
+	return resBody.Data, nil
+}
+
+func (s *SensorBucketAPI) ListPipelines(ids []uuid.UUID) ([]processing.Pipeline, error) {
+	q := url.Values{}
+	q["id"] = lo.Map(ids, func(id uuid.UUID, _ int) string { return id.String() })
```
[the remark this answers is not in the document]
Comment (TimVosch) [reply services/dashboard/infra/api_stores.go:66]: I don't think it matters as it is only for filtering.

File: services/dashboard/main.go [hunk 5]
```diff
@@ -13,41 +13,54 @@ import (
...
+		fileServer := http.FileServer(http.Dir(STATIC_PATH))
 		router.Handle("/static/*", http.StripPrefix("/static", fileServer))
 	}
 
+	sbAPI := dashboardinfra.NewSensorBucketAPI(EP_INGRESSES, EP_PIPELINES, EP_TRACES, EP_DEVICES)
+	// tracesMock := dashboardinfra.NewTracesMock()
+
 	router.Get("/", func(w http.ResponseWriter, r *http.Request) { http.Redirect(w, r, "/overview", http.StatusFound) })
-	router.Mount("/overview", createOverviewPageHandler())
+	router.Mount("/overview", routes.CreateOverviewPageHandler())
```
[the remark this answers is not in the document]
Comment (TimVosch) [reply services/dashboard/main.go:62]: get is for a GET request on an endpoint. Wheras Mount mounts a router at a subpath

File: services/tracing/migrations/20230821113939_initial_tracing.up.sql (hunk 1 above)
Comment (TimVosch) [reply services/tracing/migrations/20230821113939_initial_tracing.up.sql:18]: Save the timestamp to 3 decimals of the second. So milliseconds. 
I noticed that the tracing service only outputs seconds, and I couldn't get it fixed. This is pretty much a remnant.

File: services/tracing/tracing/infra/store_psql.go (hunk 2 above)
Comment (TimVosch) [reply services/tracing/tracing/infra/store_psql.go:107]: If device_id is NULL it instead returns 0. Allowing the trace service to just `lo.Max(tace.steps.deviceID)` to get the device ID. As well as saving some null pointer checks.

**Review by JEFFTheDev — APPROVED** [review]

**Pull request merged**